### ISSUE TITLE
no-restricted-syntax rule doesn't report TSEnumDeclaration, but shows an error after running yarn lint

### ISSUE THREAD

**Issue by ntorbinskiy** [eslint/eslint#17608]
### Environment

Node version: v18.16.1
npm version: v9.5.1
Local ESLint version: v8.50.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.22621

### What parser are you using?

Default (Espree)

### What did you do?

I needed to add rule, which will report a problem, when I use typescript `enum` in project. I made everything by [this](https://github.com/typescript-eslint/typescript-eslint/issues/561#issuecomment-496664453) comment, but it works only when I run eslint

code: 
```typescript
"no-restricted-syntax": [
      "error",
      // other restrictions
      {
        "selector": "TSEnumDeclaration",
        "message": "Don't declare  enums, use unions"
      }
    ]
```

demo of the problem: https://www.loom.com/share/4acc28e106514c8d919e54127de9015d

### What did you expect to happen?

When I type enum in my code - report an error

### What actually happened?

In my project: 
* I have error about my enum in terminal after running `yarn lint` command
* I don't have any error when I check code withut running `yarn lint`

At playground: 
This rule works okay, and reports an error without problems

### Link to Minimal Reproducible Example

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.ts&code=KYOwrgtgBAglDeBYAUFWKC%2BQ&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaaFAF2gEsBjUxAE0OQE9dSBDAD3TAG1xsciaNHzRIAGn4CsA7JGSIkNUd0gBZRAFsARkICiHAA4lkyCvlw982gFaIaAOlxtNiALwByYxWSa2AYX8PAF0eY3xDIVImJxd3DwASAEdYISYAJTYAdwBVXGQ2ADNEEIkpWUhXUzYAc0RVAEEAN3wKOjBYM1wasGTU6AzsvILisDZScgptWFowQpFNMAAFSl82B0hyiABfSVlMLbkFJVIVDHUtXWgDYxQzCytbe1JY109vNf82eCpEJgB9HIAOQAyg0AGJ6EJhESRaDRV7xPppTK5fJFErBMr7HBVAp1RotNodLo9ZEDVHDDFjCaUaazeb4RYrHx%2BDaHXaHGT7eSKZ5nKAAFRBelwsE0ABF7PA2NBxuZcGgCKQABQ8KgWZCkNzkVLBACU2JxlTutXq5wlFg8pDAdGlssQYAIuEIGvyNsQYs0qA55WC-G2IG2QA&tsconfig=N4KABARAxg9gtgBwJYBsCmAnA8ggLkmAOwGcIAuMUMSAIwENi0BVDFcyAOggBpxJc6GAOZpc7CGmIA2HnwgokNdgG0%2B1CABN4s6uq1wOSXJjo10O3ROKE0ADzF8Aur3V0UKGAHcAUqQq4MAFc0F0hiAGskBAAZRQBhAAs0KHD2AODQiGIApCgxfyCQuQAzGAwoNDiiYiRstEJcOIYkQiEASUIAMVQ0ADk6OEk0wszCGABROCNhjLlJAFkYDUD0DuMMGAQZovU4JZW0cUkbewsIPeX0ACVJGBRA-CJxMY1DzIxblAA3NF8iRcuhwKs3UtTudGMGgBBz8YHSO0gACtiLZxAgPowMD8zi0oB9Bg03NsQABfXgkoA&tokens=false

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by ntorbinskiy** [comment]
omg, my eslint extension in VS code was turned off, that's why error wasn't reported😂 

Closing this issue
